dotCMS/src/main/java/com/dotcms/storage/FileStorageAPIImpl.java

```diff
@@ -0,0 +1,308 @@
+    }
+
+    private void checkOverride (final Storage storage,
+                                final GenerateMetaDataConfiguration generateMetaDataConfiguration) {
+
+        final StorageKey storageKey = generateMetaDataConfiguration.getStorageKey();
+        if (generateMetaDataConfiguration.isOverride() && storage.existsObject(storageKey.getGroup(), storageKey.getKey())) {
+
+            try {
+
+                storage.deleteObject(storageKey.getGroup(), storageKey.getKey());
+            } catch (Exception e) {
+
+                Logger.error(this.getClass(),
+                        String.format("Unable to delete existing metadata file [%s] [%s]",
+                                storageKey.getKey(), e.getMessage()), e);
+            }
+        }
+    } // checkOverride.
+
+    @Override
+    public Map<String, Object> retrieveMetaData(final RequestMetaData requestMetaData) {
+
+        Map<String, Object> metadataMap = Collections.emptyMap();
+
+        if (requestMetaData.isCache()) {
+
+            final DotCacheAdministrator cacheAdmin = CacheLocator.getCacheAdministrator();
+            if (null != cacheAdmin) {
+
+                metadataMap = Try.of(()->(Map<String, Object>)cacheAdmin.get(requestMetaData.getCacheKeySupplier().get(),
+                        CACHE_GROUP)).getOrElse(Collections.emptyMap());
+            }
+        }
+
+        if (!UtilMethods.isSet(metadataMap)) {
+
+            final StorageKey    storageKey     = requestMetaData.getStorageKey();
+            final String        storageType    = this.getStorageType (storageKey);
+            final Storage       storage        = this.getStorageProvider().getStorage(storageType);
+
+            this.checkBucket(storageKey, storage);
+            if (storage.existsObject(storageKey.getGroup(), storageKey.getKey())) {
+
+                metadataMap =  this.retrieveMetadata(storageKey, storage);
+                Logger.info(this, "Retrieve the meta data from storage, path: " + storageKey.getKey());
+                if (null != requestMetaData.getCacheKeySupplier()) {
+                    this.putIntoCache(requestMetaData.getCacheKeySupplier().get(),
+                            requestMetaData.getWrapMetadataMapForCache().apply(metadataMap));
+                }
+            }
+        } else {
+
+            Logger.info(this, "Retrieve the meta data from cache, key: " + requestMetaData.getCacheKeySupplier().get());
+        }
+
+        return metadataMap;
+    }
+
+    private class TikaMetadataGenerator implements MetadataGenerator {
+
+        @Override
+        public Map<String, Object> generate(final File binary, final long maxLength) {
+
+            try {
+
+                final TikaUtils tikaUtils = new TikaUtils();
+                final Map<String, Object> tikaMetaDataMap = tikaUtils.getForcedMetaDataMap(binary,  Long.valueOf(maxLength).intValue());
+                return tikaMetaDataMap;
```

Comment:
Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'tikaMetaDataMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)